tests/conftest.py

```diff
@@ -27,14 +30,42 @@ def manager():
             os.chdir(oldDirectory)
     return manager
 
-## Command-line option to skip very slow tests
 
 def pytest_addoption(parser):
+    # option to skip very slow tests
     parser.addoption('--fast', action='store_true', help='skip very slow tests')
+    # option to skip parser generation
+    parser.addoption('--skip-pegen', action='store_true', help='generate the parser before running tests')
+
+class ParserGenerationException(BaseException):
+    pass
 
 def pytest_configure(config):
     config.addinivalue_line('markers', 'slow: mark test as very slow')
 
+    if not config.getoption("skip_pegen"):
+        projectRootDir = Path(__file__).parent.parent
+        syntaxDir = projectRootDir / "src" / "scenic" / "syntax"
+        grammar = syntaxDir / "scenic.gram"
+        parser = syntaxDir / "parser.py"
+        result = subprocess.run(
+            [
+                "poetry",
+                "run",
+                "python",
+                "-m",
+                "pegen",
+                grammar,
+                "-o",
+                parser,
+            ],
+            cwd=projectRootDir,
+            capture_output=True,
+            text=True,
+        )
+        if result.returncode != 0:
+            raise ParserGenerationException(f"Failed to generate the parser: {result.stderr}")
```

Comment:
Any reason for a new kind of exception here? We could just use `pytest.exit`.

Reply:
Fixed faec3f12e3481185a78e9be1eaf7f944ecbbdd31

Reply:
Fixed in c204e3c8de2dc15d60bcb517ad6cb4efd4b96e52

Reply:
Let's set the return code too: `pytest.exit` uses 0 by default, whereas we should return 1 to indicate an error or even propagate the value from the subprocess.

Reply:
I believe propagating the value from the subprocess makes sense and implemented it in c098877b505a5225ddb70fac990af0c4715730f3 (It seems that pegen always returns `1` for errors, making it technically the same as setting return code 1 for `pytest.exit`)

<img width="743" alt="Screen Shot 2022-09-21 at 9 37 55" src="https://user-images.githubusercontent.com/10496155/191562238-54ef9bed-8e93-40ac-9545-34c37676ff61.png">

I tested my code against invalid grammar and confirmed that the status code is `1` after execution.